exchanges/bybit/bybit_wrapper.go

```diff
@@ -0,0 +1,2044 @@
+		}
+
+		if len(resp) != 1 {
+			return order.Detail{}, fmt.Errorf("%w, received %v orders", errExpectedOneOrder, len(resp))
+		}
+
+		return order.Detail{
+			Amount:         resp[0].Qty,
+			Exchange:       by.Name,
+			OrderID:        resp[0].OrderID,
+			ClientOrderID:  resp[0].OrderLinkID,
+			Side:           getSide(resp[0].Side),
+			Type:           getTradeType(resp[0].OrderType),
+			Pair:           pair,
+			Cost:           resp[0].CumulativeQty,
+			AssetType:      assetType,
+			Status:         getOrderStatus(resp[0].OrderStatus),
+			Price:          resp[0].Price,
+			ExecutedAmount: resp[0].Qty - resp[0].LeavesQty,
+			Date:           resp[0].CreatedAt,
+			LastUpdated:    resp[0].UpdatedAt,
+		}, nil
+
+	case asset.Futures:
+		resp, err := by.GetActiveRealtimeOrders(ctx, pair, orderID, "")
+		if err != nil {
+			return order.Detail{}, err
+		}
+
+		if len(resp) != 1 {
+			return order.Detail{}, fmt.Errorf("%w, received %v orders", errExpectedOneOrder, len(resp))
+		}
+
+		return order.Detail{
+			Amount:         resp[0].Qty,
+			Exchange:       by.Name,
+			OrderID:        resp[0].OrderID,
+			ClientOrderID:  resp[0].OrderLinkID,
+			Side:           getSide(resp[0].Side),
+			Type:           getTradeType(resp[0].OrderType),
+			Pair:           pair,
+			Cost:           resp[0].CumulativeQty,
+			AssetType:      assetType,
+			Status:         getOrderStatus(resp[0].OrderStatus),
+			Price:          resp[0].Price,
+			ExecutedAmount: resp[0].Qty - resp[0].LeavesQty,
+			Date:           resp[0].CreatedAt,
+			LastUpdated:    resp[0].UpdatedAt,
+		}, nil
+
+	case asset.USDCMarginedFutures:
+		resp, err := by.GetActiveUSDCOrder(ctx, pair, "PERPETUAL", orderID, "", "", "", "", 0)
+		if err != nil {
+			return order.Detail{}, err
+		}
+
+		if len(resp) != 1 {
+			return order.Detail{}, fmt.Errorf("%w, received %v orders", errExpectedOneOrder, len(resp))
+		}
+
+		return order.Detail{
+			Amount:         resp[0].Qty,
+			Exchange:       by.Name,
+			OrderID:        resp[0].ID,
+			ClientOrderID:  resp[0].OrderLinkID,
+			Side:           getSide(resp[0].Side),
+			Type:           getTradeType(resp[0].OrderType),
+			Pair:           pair,
+			Cost:           resp[0].TotalOrderValue,
+			AssetType:      assetType,
+			Status:         getOrderStatus(resp[0].OrderStatus),
+			Price:          resp[0].Price,
+			ExecutedAmount: resp[0].TotalFilledQty,
+			Date:           resp[0].CreatedAt.Time(),
+		}, nil
+
+	default:
+		return order.Detail{}, fmt.Errorf("%s %w", assetType, asset.ErrNotSupported)
+	}
+}
+
+// GetDepositAddress returns a deposit address for a specified currency
+func (by *Bybit) GetDepositAddress(ctx context.Context, cryptocurrency currency.Code, _, chain string) (*deposit.Address, error) {
+	dAddressInfo, err := by.GetDepositAddressForCurrency(ctx, cryptocurrency.String())
+	if err != nil {
+		return nil, err
+	}
+
+	for x := range dAddressInfo.Chains {
+		if dAddressInfo.Chains[x].Chain == chain {
```

Comment:
This is broken. 

```
[ERROR] | LOG | 01/08/2022 09:46:23 | Bybit failed to get cryptocurrency available transfer chains. Err: unsupported wrapper function
```

Reply:
You have the following in the wrapper:
```go
				MultiChainDeposits:    true,
				MultiChainWithdrawals: true,
```
however, to support that, you need an implementation of the `interfaces.go` function `GetAvailableTransferChains(ctx context.Context, cryptocurrency currency.Code) ([]string, error)`

Reply:
Okay, @gloriousCode for the above explanation. It's helpful.

Reply:
So by default, the function that calls this under `engine\helper.go` in  the function `GetAllExchangeCryptocurrencyDepositAddresses` sends `chain` as `""`. So this function will need to handle blank chain supplied without error. As currently this is outputting:
```
[ERROR] | LOG | 02/08/2022 11:04:39 | Bybit failed to get cryptocurrency deposit address for USDT. Err: deposit address not found for currency: USDT chain: 
```

Because for bybit, `RequiresChainSet` is false and there is not chain called `USDT`. If `chain == ""`, then you can return the first chain aka

```go
	dAddressInfo, err := by.GetDepositAddressForCurrency(ctx, cryptocurrency.String())
	if err != nil {
		return nil, err
	}

	for x := range dAddressInfo.Chains {
		if dAddressInfo.Chains[x].Chain == chain || chain == "" {
			return &deposit.Address{
				Address: dAddressInfo.Chains[x].DepositAddress,
				Tag:     dAddressInfo.Chains[x].DepositTag,
				Chain:   dAddressInfo.Chains[x].Chain,
			}, nil
		}
	}
```

Reply:
Okay